### ISSUE TITLE
Deferred chart creation

### ISSUE THREAD

**Issue by denchen** [ahmadsoe/ember-highcharts#40]
Let me preface this by saying I'm still quite new to Ember, though I have prior experience with HighCharts, so this may seem like an easy question to solve (it may not even be specific to HighCharts). I was able to get the example code working just fine. Now I am trying to figure out the best approach to what I'm trying to do.

I have a bunch of chart data. When the page first renders, there won't be any charts rendered. I want to render a chart based on the user's selection (eg. from a dropdown or elsewhere on the page). I don't need to go back and load more data; the data is already loaded. How should I extend high-charts to render a new chart based on what's selected?

In other words, I basically want

    {{my-high-charts allChartData=allChartData selectedChart=selectedChart chartOptions=chartOptions}}

Where initially the bound property `selectedChart` is null. When `selectedChart` is updated, I'll know what data in `allChartData` to use, and then I can instantiate a HighCharts. And if `selectedChart` changes again, it replaces the old chart either by updating it or destroying/recreating the chart (whichever makes more sense). So what is the best way to go about this?

**Comment by ahmadsoe** [comment]
You can use `{{component}}` helper to render component dynamically. 

```{{component selectedChart chartOptions=chartOptions .... }}```

more info about this helper is here http://emberjs.com/api/classes/Ember.Templates.helpers.html#method_component

**Comment by denchen** [comment]
Thanks. What I ended up doing is using the component helper to initially render an empty component, and then when selectedChart is set to non-null, I render my dynamic-high-chart component. Probably not that elegant or even the proper way to do things, but it works.

That being said, I'm still a little confused as to how Ember components get rendered / re-rendered. Let's say I have a chart already rendered. But then I completely replace `chartData` that's bound to `content`. Shouldn't the entire component be re-rendered, according to this:

http://emberjs.com/blog/2015/05/24/another-two-oh-status-update.html#toc_new-lifecycle-hooks

If that is the case, and the entire component is re-rendered, why do we need the `contentDidChange()` function that's in the example in the `README`, since the chart gets re-created anyway? Or am I completely misunderstanding what "rerendering" in Ember actually is? (My take on it is the component (HighCharts) is destroyed and re-created with the new attributes)

**Comment by ahmadsoe** [comment]
Updating `chartData` that's bound to `content` will not re-render the entire component, it just to update the chart series (like x or y values). 
The `contentDidChange()` in the readme is for a special case if you want to also update another chart properties (e.g. chart title, sub-title, axis label) via HighCharts API.

**Comment by gregjopa** [comment]
This questions seems to be answered. I think this issue can be closed.
